### PR TITLE
Sync recent NS2.1 changes

### PR DESCRIPTION
**Pull request by EgorBo**
1) Copied changes from corefx for System.Data (also https://github.com/mono/corefx/pull/310)
2) New Tracing api is left empty (because the current API is empty too)
3) New AES types throw PNSE

Bump netstandard.dll AssemblyVersion to 2.1 (fixes https://github.com/mono/mono/issues/13412)

### CODE REVIEW

**Comment by marek-safar** [comment]
Please don't forget to backport to 2019-06

File: mcs/class/corlib/corefx/AesCcm.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public sealed partial class AesCcm : System.IDisposable
+    {
+        public AesCcm (byte[] key) => throw new PlatformNotSupportedException ();
+        public AesCcm (System.ReadOnlySpan<byte> key) => throw new PlatformNotSupportedException ();
+        public static System.Security.Cryptography.KeySizes NonceByteSizes => throw new PlatformNotSupportedException ();
+        public static System.Security.Cryptography.KeySizes TagByteSizes => throw new PlatformNotSupportedException ();
+        public void Decrypt (byte[] nonce, byte[] ciphertext, byte[] tag, byte[] plaintext, byte[] associatedData = null) => throw new PlatformNotSupportedException ();
+        public void Decrypt (System.ReadOnlySpan<byte> nonce, System.ReadOnlySpan<byte> ciphertext, System.ReadOnlySpan<byte> tag, System.Span<byte> plaintext, System.ReadOnlySpan<byte> associatedData = default(System.ReadOnlySpan<byte>)) => throw new PlatformNotSupportedException ();
+        public void Dispose () => throw new PlatformNotSupportedException ();
```
Comment (akoeplinger) [inline mcs/class/corlib/corefx/AesCcm.cs]: `Dispose()` must not throw an exception (this causes issues like https://github.com/mono/mono/commit/fa21013d7f071c9746d55bb976135940762acb8a).

File: mcs/class/corlib/corefx/AesCcm.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class AesCcm : System.IDisposable
+    {
+        public AesCcm (byte[] key) => throw new PlatformNotSupportedException ();
```
Comment (akoeplinger) [inline mcs/class/corlib/corefx/AesCcm.cs]: should we throw `NotImplementedException` instead?

File: mcs/class/corlib/corefx/AesGcm.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public sealed partial class AesGcm : System.IDisposable
+    {
+        public AesGcm (byte[] key) => throw new PlatformNotSupportedException ();
+        public AesGcm (System.ReadOnlySpan<byte> key) => throw new PlatformNotSupportedException ();
+        public static System.Security.Cryptography.KeySizes NonceByteSizes => throw new PlatformNotSupportedException ();
+        public static System.Security.Cryptography.KeySizes TagByteSizes => throw new PlatformNotSupportedException ();
+        public void Decrypt (byte[] nonce, byte[] ciphertext, byte[] tag, byte[] plaintext, byte[] associatedData = null) => throw new PlatformNotSupportedException ();
+        public void Decrypt (System.ReadOnlySpan<byte> nonce, System.ReadOnlySpan<byte> ciphertext, System.ReadOnlySpan<byte> tag, System.Span<byte> plaintext, System.ReadOnlySpan<byte> associatedData = default(System.ReadOnlySpan<byte>)) => throw new PlatformNotSupportedException ();
+        public void Dispose () => throw new PlatformNotSupportedException ();
```
Comment (akoeplinger) [inline mcs/class/corlib/corefx/AesGcm.cs]: same here.

File: mcs/class/corlib/corefx/AesCcm.cs (hunk 2 above)
Comment (steveisok) [reply mcs/class/corlib/corefx/AesCcm.cs]: Correct me if I am wrong, but I believe our previous cryptography adds threw PNSE's.

**Comment by steveisok** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff and csproj

**Comment by akoeplinger** [comment]
@monojenkins backport 2019-06

**Comment by ognamala** [comment]
AESGCM Support is missing there is already the following open issue [https://github.com/mono/mono/issues/19285](url)

**Pull request merged**